core/observe.c

```diff
@@ -1493,4 +1493,33 @@ bool observe_handleNotify(lwm2m_context_t * contextP,
     }
     return true;
 }
+
+#ifndef LWM2M_VERSION_1_0
+uint8_t reporting_handleSend(lwm2m_context_t *contextP, void *fromSessionH, coap_packet_t *message) {
+    lwm2m_client_t *clientP;
+    lwm2m_media_type_t format;
+
+    LOG_DBG("Entering");
+
+    if (message->code != COAP_POST)
+        return COAP_400_BAD_REQUEST;
+
+    for (clientP = contextP->clientList; clientP != NULL; clientP = clientP->next) {
+        if (clientP->sessionH == fromSessionH)
+            break;
+    }
+    if (clientP == NULL)
+        return COAP_400_BAD_REQUEST;
+
+    format = utils_convertMediaType(message->content_type);
+
+    if (contextP->monitorCallback != NULL) {
```

Comment:
I would rather create a separate callback with an interface like
```
void lwm2m_reporting_set_send_callback(lwm2m_context_t *contextP, lwm2m_result_callback_t callback, void *userData);
```
to report send messages. The `monitorCallback` should only be used for the "Client Registration Interface" and not for the "Information Reporting Interface". Also I would put this in a separate file like `core/reporting.c`.

Reply:
I made the change.